core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java

```diff
@@ -57,6 +59,8 @@
   private final UnderFileSystem mUnderFileSystem;
   private final UnderFileSystemConfiguration mConf;
   private final String mPath;
+  private final Supplier<String> mEscapedPath =
+      CommonUtils.memoize(this::escapePath);
```

Comment:
what does memoizing do for us here?

Reply:
So that when MetricsSystem record the metrics of any UFS operation,  the escaped path can be memorized without constructing Alluxio URI and call metho in MetricsSystem.

Reply:
Isn't the escaped path always the same though because `mPath` is final?

Reply:
Make sense, Changed to initialize directly in the constructor.